src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figure = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        for (int i = 0; i < 3; i++) {
+            figure[i] = figureSupplier.getRandomFigure();
```

Comment:
The 'setColor' method seems to be called immediately after getting a random figure. It might be a good idea to set the color within the 'getRandomFigure' method itself to encapsulate the behavior related to creating a fully initialized figure.